plugin/semantic.py

```diff
@@ -0,0 +1,232 @@
+NON_MODIFIER_SCOPES = {
+            'variable'     :'variable.other.lsp',
+            'parameter'    :'variable.parameter.lsp',
+            'function'     :'variable.function.lsp',
+            'method'       :'variable.function.lsp',
+            'property'     :'variable.other.member.lsp',
+            'class'        :'entity.name.class.lsp',
+            'enum'         :'variable.enum.lsp',
+            'enumMember'   :'entity.name.enum.lsp',
+            'type'         :'storage.type.lsp',
+            'macro'        :'variable.other.constant.lsp',
+            'namespace'    :'variable.other.namespace.lsp',
+            'typeParameter':'variable.parameter.generic.lsp',
+            'comment'      :'comment.block.documentation.lsp',
+            'dependent'    :'',
+            'concept'      :'',
+            'module'       :'',
+            'magicFunction':'',
+            'selfParameter':'',
+
+            }
+
+DECLARATION_SCOPES = {
+            'namespace':'entity.name.namespace.lsp',
+            'type'     :'entity.name.type.lsp',
+            'class'    :'entity.name.class.lsp',
+            'enum'     :'entity.name.enum.lsp',
+            'interface':'entity.name.interface.lsp',
+            'struct'   :'entity.name.struct.lsp',
+            'function' :'entity.name.function.lsp',
+            'method'   :'entity.name.function.lsp',
+            'macro'    :'entity.name.macro.lsp'
+            }
+
+STAIC_SCOPES        = NON_MODIFIER_SCOPES.copy()
+DEPRECATED_SCOPES   = NON_MODIFIER_SCOPES.copy()
+ABSTRACT_SCOPES     = NON_MODIFIER_SCOPES.copy()
+ASYNC_SCOPES        = NON_MODIFIER_SCOPES.copy()
+MODIFICATION_SCOPES = NON_MODIFIER_SCOPES.copy()
+DEFAULT_LIB_SCOPES  = NON_MODIFIER_SCOPES.copy()
+
+[NON_MODIFIER_SCOPES.update({k:v + ', semantic.'+ k + '.lsp'})          for k,v in NON_MODIFIER_SCOPES.items()]
+
+[DECLARATION_SCOPES.update( {k:v + ', semantic.'+ k + '.declaration.lsp'})    for k,v in DECLARATION_SCOPES.items()]
+[STAIC_SCOPES.update(       {k:v + ', semantic.'+ k + '.satic.lsp'})          for k,v in STAIC_SCOPES.items()]
+[DEPRECATED_SCOPES.update(  {k:v + ', semantic.'+ k + '.deprecated.lsp'})     for k,v in DEPRECATED_SCOPES.items()]
+[ABSTRACT_SCOPES.update(    {k:v + ', semantic.'+ k + '.abstract.lsp'})       for k,v in ABSTRACT_SCOPES.items()]
+[ASYNC_SCOPES.update(       {k:v + ', semantic.'+ k + '.async.lsp'})          for k,v in ASYNC_SCOPES.items()]
+[MODIFICATION_SCOPES.update({k:v + ', semantic.'+ k + '.modification.lsp'})   for k,v in MODIFICATION_SCOPES.items()]
+[DEFAULT_LIB_SCOPES.update( {k:v + ', semantic.'+ k + '.defaultLibrary.lsp'}) for k,v in DEFAULT_LIB_SCOPES.items()]
+
+SEMANTIC_SCOPES = {
+            ''              :NON_MODIFIER_SCOPES,# if no modifiers are provided
+
+            "declaration"   :DECLARATION_SCOPES,
+            "definition"    :DECLARATION_SCOPES,
+
+            "readonly"      :{'variable': 'constant.other.lsp, variable.other.constant.lsp, semantic.variable.readonly.lsp'},
```

Comment:
This should only use one of the scopes.

Reply:
@jwortmann 
you mean this one:
`"readonly" :{'variable': 'constant.other.lsp, variable.other.constant.lsp, semantic.variable.readonly.lsp'},`?